### ISSUE TITLE
VPC configuration in cluster.yaml is not honored by kube-aws

### ISSUE THREAD

**Issue by rclayton-the-terrible** [coreos/coreos-kubernetes#259]
```yaml
# Unique name of Kubernetes cluster. In order to deploy
# more than one cluster into the same AWS account, this
# name must not conflict with an existing cluster.
clusterName: kubernetes-example

# Name of the SSH keypair already loaded into the AWS
# account being used to deploy this cluster.
keyName: example

# Region to provision Kubernetes cluster
region: us-west-2

# Availability Zone to provision Kubernetes cluster
#availabilityZone:

# DNS name routable to the Kubernetes controller nodes
# from worker nodes and external clients. The deployer
# is responsible for making this name routable
externalDNSName: kube.example.com

# Instance type for controller node
#controllerInstanceType: m3.medium

# Disk size (GiB) for controller node
#controllerRootVolumeSize: 30

# Number of worker nodes to create
workerCount: 3

# Instance type for worker nodes
#workerInstanceType: m3.medium

# Disk size (GiB) for worker nodes
#workerRootVolumeSize: 30

# Location of kube-aws artifacts used to deploy a new
# Kubernetes cluster. The necessary artifacts are already
# available in a public S3 bucket matching the version
# of the kube-aws tool. This parameter is typically
# overwritten only for development purposes.
#artifactURL: https://coreos-kubernetes.s3.amazonaws.com/<VERSION>

# CIDR for Kubernetes VPC
vpcCIDR: "10.3.0.0/16"

# CIDR for Kubernetes subnet
instanceCIDR: "10.3.0.0/24"

# IP Address for controller in Kubernetes subnet
controllerIP: 10.3.0.50

# CIDR for all service IP addresses
serviceCIDR: "10.5.0.0/24"

# CIDR for all pod IP addresses
podCIDR: "10.4.0.0/16"

# IP address of Kubernetes controller service (must be contained by serviceCIDR)
kubernetesServiceIP: 10.5.0.1

# IP address of Kubernetes dns service (must be contained by serviceCIDR)
dnsServiceIP: 10.5.0.10
```
Then running the command:  `./kube-aws --config="cluster.yaml" render`

I get the following output:

```json
{
    "AWSTemplateFormatVersion": "2010-09-09",
    "Conditions": {
        "EmptyAvailabilityZone": {
            "Fn::Equals": [
                {
                    "Ref": "AvailabilityZone"
                },
                ""
            ]
        }
    },
    "Description": "kube-aws Kubernetes cluster",
    "Mappings": {
        "RegionMap": {
            "ap-northeast-1": {
                "alpha": "ami-208db24e",
                "beta": "ami-13f2cc7d"
            },
            "ap-southeast-1": {
                "alpha": "ami-d830fcbb",
                "beta": "ami-0e66aa6d"
            },
            "ap-southeast-2": {
                "alpha": "ami-b3af8ad0",
                "beta": "ami-cd90b5ae"
            },
            "eu-central-1": {
                "alpha": "ami-b22138de",
                "beta": "ami-e4988188"
            },
            "eu-west-1": {
                "alpha": "ami-2cf84d5f",
                "beta": "ami-f961d48a"
            },
            "sa-east-1": {
                "alpha": "ami-602aab0c",
                "beta": "ami-bfc242d3"
            },
            "us-east-1": {
                "alpha": "ami-4b240221",
                "beta": "ami-0c082966"
            },
            "us-gov-west-1": {
                "alpha": "ami-cbfc40aa",
                "beta": "ami-baf448db"
            },
            "us-west-1": {
                "alpha": "ami-d5afdbb5",
                "beta": "ami-72eb9f12"
            },
            "us-west-2": {
                "alpha": "ami-996386f9",
                "beta": "ami-66e30706"
            }
        }
    },
    "Parameters": {
        "APIServerCert": {
            "Description": "PEM-formatted kube-apiserver certificate, base64-encoded",
            "Type": "String"
        },
        "APIServerKey": {
            "Description": "PEM-formatted kube-apiserver key, base64-encoded",
            "Type": "String"
        },
        "ArtifactURL": {
            "Default": "https://coreos-kubernetes.s3.amazonaws.com/v0.3.0",
            "Description": "Public location of coreos-kubernetes deployment artifacts",
            "Type": "String"
        },
        "AvailabilityZone": {
            "Default": "",
            "Description": "Specific availability zone (optional)",
            "Type": "String"
        },
        "CACert": {
            "Description": "PEM-formattd CA certificate, base64-encoded",
            "Type": "String"
        },
        "ClusterName": {
            "Default": "kubernetes",
            "Description": "Name of Kubernetes cluster",
            "Type": "String"
        },
        "ControllerInstanceType": {
            "Default": "m3.medium",
            "Description": "EC2 instance type used for each controller instance",
            "Type": "String"
        },
        "ControllerRootVolumeSize": {
            "Default": "30",
            "Description": "Controller root volume size (GiB)",
            "Type": "String"
        },
        "KeyName": {
            "Description": "Name of SSH keypair to authorize on each instance",
            "Type": "String"
        },
        "ReleaseChannel": {
            "AllowedValues": [
                "alpha",
                "beta"
            ],
            "Default": "alpha",
            "Description": "CoreOS Linux release channel to use as instance operating system",
            "Type": "String"
        },
        "WorkerCert": {
            "Description": "PEM-formatted kubelet (worker) certificate, base64-encoded",
            "Type": "String"
        },
        "WorkerCount": {
            "Default": "1",
            "Description": "Number of worker instances to create, may be modified later",
            "Type": "String"
        },
        "WorkerInstanceType": {
            "Default": "m3.medium",
            "Description": "EC2 instance type used for each worker instance",
            "Type": "String"
        },
        "WorkerKey": {
            "Description": "PEM-formatted kubelet (worker) key, base64-encoded",
            "Type": "String"
        },
        "WorkerRootVolumeSize": {
            "Default": "30",
            "Description": "Worker root volume size (GiB)",
            "Type": "String"
        }
    },
    "Resources": {
        "AlarmControllerRecover": {
            "Properties": {
                "AlarmActions": [
                    {
                        "Fn::Join": [
                            "",
                            [
                                "arn:aws:automate:",
                                {
                                    "Ref": "AWS::Region"
                                },
                                ":ec2:recover"
                            ]
                        ]
                    }
                ],
                "AlarmDescription": "Trigger a recovery when system check fails for 5 consecutive minutes.",
                "ComparisonOperator": "GreaterThanThreshold",
                "Dimensions": [
                    {
                        "Name": "InstanceId",
                        "Value": {
                            "Ref": "InstanceController"
                        }
                    }
                ],
                "EvaluationPeriods": "5",
                "MetricName": "StatusCheckFailed_System",
                "Namespace": "AWS/EC2",
                "Period": "60",
                "Statistic": "Minimum",
                "Threshold": "0"
            },
            "Type": "AWS::CloudWatch::Alarm"
        },
        "AutoScaleWorker": {
            "Properties": {
                "AvailabilityZones": [
                    {
                        "Fn::If": [
                            "EmptyAvailabilityZone",
                            {
                                "Fn::Select": [
                                    "0",
                                    {
                                        "Fn::GetAZs": {
                                            "Ref": "AWS::Region"
                                        }
                                    }
                                ]
                            },
                            {
                                "Ref": "AvailabilityZone"
                            }
                        ]
                    }
                ],
                "DesiredCapacity": {
                    "Ref": "WorkerCount"
                },
                "HealthCheckGracePeriod": 600,
                "HealthCheckType": "EC2",
                "LaunchConfigurationName": {
                    "Ref": "LaunchConfigurationWorker"
                },
                "MaxSize": {
                    "Ref": "WorkerCount"
                },
                "MinSize": {
                    "Ref": "WorkerCount"
                },
                "Tags": [
                    {
                        "Key": "KubernetesCluster",
                        "PropagateAtLaunch": "true",
                        "Value": {
                            "Ref": "ClusterName"
                        }
                    },
                    {
                        "Key": "Name",
                        "PropagateAtLaunch": "true",
                        "Value": "kube-aws-worker"
                    }
                ],
                "VPCZoneIdentifier": [
                    {
                        "Ref": "Subnet"
                    }
                ]
            },
            "Type": "AWS::AutoScaling::AutoScalingGroup"
        },
        "EIPController": {
            "Properties": {
                "Domain": "vpc",
                "InstanceId": {
                    "Ref": "InstanceController"
                }
            },
            "Type": "AWS::EC2::EIP"
        },
        "IAMInstanceProfileController": {
            "Properties": {
                "Path": "/",
                "Roles": [
                    {
                        "Ref": "IAMRoleController"
                    }
                ]
            },
            "Type": "AWS::IAM::InstanceProfile"
        },
        "IAMInstanceProfileWorker": {
            "Properties": {
                "Path": "/",
                "Roles": [
                    {
                        "Ref": "IAMRoleWorker"
                    }
                ]
            },
            "Type": "AWS::IAM::InstanceProfile"
        },
        "IAMRoleController": {
            "Properties": {
                "AssumeRolePolicyDocument": {
                    "Statement": [
                        {
                            "Action": [
                                "sts:AssumeRole"
                            ],
                            "Effect": "Allow",
                            "Principal": {
                                "Service": [
                                    "ec2.amazonaws.com"
                                ]
                            }
                        }
                    ],
                    "Version": "2012-10-17"
                },
                "Path": "/",
                "Policies": [
                    {
                        "PolicyDocument": {
                            "Statement": [
                                {
                                    "Action": "ec2:*",
                                    "Effect": "Allow",
                                    "Resource": "*"
                                },
                                {
                                    "Action": "elasticloadbalancing:*",
                                    "Effect": "Allow",
                                    "Resource": "*"
                                }
                            ],
                            "Version": "2012-10-17"
                        },
                        "PolicyName": "root"
                    }
                ]
            },
            "Type": "AWS::IAM::Role"
        },
        "IAMRoleWorker": {
            "Properties": {
                "AssumeRolePolicyDocument": {
                    "Statement": [
                        {
                            "Action": [
                                "sts:AssumeRole"
                            ],
                            "Effect": "Allow",
                            "Principal": {
                                "Service": [
                                    "ec2.amazonaws.com"
                                ]
                            }
                        }
                    ],
                    "Version": "2012-10-17"
                },
                "Path": "/",
                "Policies": [
                    {
                        "PolicyDocument": {
                            "Statement": [
                                {
                                    "Action": "ec2:Describe*",
                                    "Effect": "Allow",
                                    "Resource": "*"
                                },
                                {
                                    "Action": "ec2:AttachVolume",
                                    "Effect": "Allow",
                                    "Resource": "*"
                                },
                                {
                                    "Action": "ec2:DetachVolume",
                                    "Effect": "Allow",
                                    "Resource": "*"
                                }
                            ],
                            "Version": "2012-10-17"
                        },
                        "PolicyName": "root"
                    }
                ]
            },
            "Type": "AWS::IAM::Role"
        },
        "InstanceController": {
            "Properties": {
                "AvailabilityZone": {
                    "Fn::If": [
                        "EmptyAvailabilityZone",
                        {
                            "Fn::Select": [
                                "0",
                                {
                                    "Fn::GetAZs": {
                                        "Ref": "AWS::Region"
                                    }
                                }
                            ]
                        },
                        {
                            "Ref": "AvailabilityZone"
                        }
                    ]
                },
                "BlockDeviceMappings": [
                    {
                        "DeviceName": "/dev/xvda",
                        "Ebs": {
                            "VolumeSize": {
                                "Ref": "ControllerRootVolumeSize"
                            }
                        }
                    }
                ],
                "IamInstanceProfile": {
                    "Ref": "IAMInstanceProfileController"
                },
                "ImageId": {
                    "Fn::FindInMap": [
                        "RegionMap",
                        {
                            "Ref": "AWS::Region"
                        },
                        {
                            "Ref": "ReleaseChannel"
                        }
                    ]
                },
                "InstanceType": {
                    "Ref": "ControllerInstanceType"
                },
                "KeyName": {
                    "Ref": "KeyName"
                },
                "NetworkInterfaces": [
                    {
                        "AssociatePublicIpAddress": false,
                        "DeleteOnTermination": true,
                        "DeviceIndex": "0",
                        "GroupSet": [
                            {
                                "Ref": "SecurityGroupController"
                            }
                        ],
                        "PrivateIpAddress": "10.0.0.50",
                        "SubnetId": {
                            "Ref": "Subnet"
                        }
                    }
                ],
                "Tags": [
                    {
                        "Key": "KubernetesCluster",
                        "Value": {
                            "Ref": "ClusterName"
                        }
                    },
                    {
                        "Key": "Name",
                        "Value": "kube-aws-controller"
                    }
                ],
                "UserData": {
                    "Fn::Base64": {
                        "Fn::Join": [
                            "",
                            [
                                "#cloud-config\ncoreos:\n  update:\n    reboot-strategy: \"off\"\n\n  flannel:\n    interface: $private_ipv4\n\n  etcd2:\n    name: controller\n    advertise-client-urls: http://$private_ipv4:2379\n    initial-advertise-peer-urls: http://$private_ipv4:2380\n    listen-client-urls: http://0.0.0.0:2379\n    listen-peer-urls: http://0.0.0.0:2380\n    initial-cluster: controller=http://$private_ipv4:2380\n\n  units:\n  - name: etcd2.service\n    command: start\n\n  - name: install-controller.service\n    command: start\n    content: |\n      [Service]\n      ExecStart=/bin/bash /tmp/install-controller.sh\n      Type=oneshot\n\nwrite_files:\n- path: /run/coreos-kubernetes/options.env\n  content: |\n    ETCD_ENDPOINTS=http://127.0.0.1:2379\n    ARTIFACT_URL=",
                                {
                                    "Ref": "ArtifactURL"
                                },
                                "\n\n- path: /tmp/install-controller.sh\n  content: |\n    #!/bin/bash\n\n    exec bash -c \"$(curl --fail --silent --show-error --location '",
                                {
                                    "Ref": "ArtifactURL"
                                },
                                "/scripts/install-controller.sh')\"\n\n- path: /etc/kubernetes/ssl/ca.pem\n  encoding: base64\n  content: ",
                                {
                                    "Ref": "CACert"
                                },
                                "\n\n- path: /etc/kubernetes/ssl/apiserver.pem\n  encoding: base64\n  content: ",
                                {
                                    "Ref": "APIServerCert"
                                },
                                "\n\n- path: /etc/kubernetes/ssl/apiserver-key.pem\n  encoding: base64\n  content: ",
                                {
                                    "Ref": "APIServerKey"
                                },
                                "\n"
                            ]
                        ]
                    }
                }
            },
            "Type": "AWS::EC2::Instance"
        },
        "InternetGateway": {
            "Properties": {
                "Tags": [
                    {
                        "Key": "KubernetesCluster",
                        "Value": {
                            "Ref": "ClusterName"
                        }
                    }
                ]
            },
            "Type": "AWS::EC2::InternetGateway"
        },
        "LaunchConfigurationWorker": {
            "Properties": {
                "BlockDeviceMappings": [
                    {
                        "DeviceName": "/dev/xvda",
                        "Ebs": {
                            "VolumeSize": {
                                "Ref": "WorkerRootVolumeSize"
                            }
                        }
                    }
                ],
                "IamInstanceProfile": {
                    "Ref": "IAMInstanceProfileWorker"
                },
                "ImageId": {
                    "Fn::FindInMap": [
                        "RegionMap",
                        {
                            "Ref": "AWS::Region"
                        },
                        {
                            "Ref": "ReleaseChannel"
                        }
                    ]
                },
                "InstanceType": {
                    "Ref": "WorkerInstanceType"
                },
                "KeyName": {
                    "Ref": "KeyName"
                },
                "SecurityGroups": [
                    {
                        "Ref": "SecurityGroupWorker"
                    }
                ],
                "UserData": {
                    "Fn::Base64": {
                        "Fn::Join": [
                            "",
                            [
                                "#cloud-config\ncoreos:\n  update:\n    reboot-strategy: \"off\"\n\n  flannel:\n    interface: $private_ipv4\n    etcd_endpoints: http://10.0.0.50:2379\n\n  units:\n  - name: install-worker.service\n    command: start\n    content: |\n      [Service]\n      ExecStart=/bin/bash /tmp/install-worker.sh\n      Type=oneshot\n\nwrite_files:\n- path: /run/coreos-kubernetes/options.env\n  content: |\n    ETCD_ENDPOINTS=http://10.0.0.50:2379\n    CONTROLLER_ENDPOINT=https://10.0.0.50\n    ARTIFACT_URL=",
                                {
                                    "Ref": "ArtifactURL"
                                },
                                "\n\n- path: /tmp/install-worker.sh\n  content: |\n    #!/bin/bash\n\n    exec bash -c \"$(curl --fail --silent --show-error --location '",
                                {
                                    "Ref": "ArtifactURL"
                                },
                                "/scripts/install-worker.sh')\"\n\n- path: /etc/kubernetes/ssl/ca.pem\n  encoding: base64\n  content: ",
                                {
                                    "Ref": "CACert"
                                },
                                "\n\n- path: /etc/kubernetes/ssl/worker.pem\n  encoding: base64\n  content: ",
                                {
                                    "Ref": "WorkerCert"
                                },
                                "\n\n- path: /etc/kubernetes/ssl/worker-key.pem\n  encoding: base64\n  content: ",
                                {
                                    "Ref": "WorkerKey"
                                },
                                "\n"
                            ]
                        ]
                    }
                }
            },
            "Type": "AWS::AutoScaling::LaunchConfiguration"
        },
        "RouteTable": {
            "Properties": {
                "Tags": [
                    {
                        "Key": "KubernetesCluster",
                        "Value": {
                            "Ref": "ClusterName"
                        }
                    }
                ],
                "VpcId": {
                    "Ref": "VPC"
                }
            },
            "Type": "AWS::EC2::RouteTable"
        },
        "RouteToInternet": {
            "Properties": {
                "DestinationCidrBlock": "0.0.0.0/0",
                "GatewayId": {
                    "Ref": "InternetGateway"
                },
                "RouteTableId": {
                    "Ref": "RouteTable"
                }
            },
            "Type": "AWS::EC2::Route"
        },
        "SecurityGroupController": {
            "Properties": {
                "GroupDescription": {
                    "Ref": "AWS::StackName"
                },
                "SecurityGroupEgress": [
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 0,
                        "IpProtocol": "tcp",
                        "ToPort": 65535
                    },
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 0,
                        "IpProtocol": "udp",
                        "ToPort": 65535
                    }
                ],
                "SecurityGroupIngress": [
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 3,
                        "IpProtocol": "icmp",
                        "ToPort": -1
                    },
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 22,
                        "IpProtocol": "tcp",
                        "ToPort": 22
                    },
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 443,
                        "IpProtocol": "tcp",
                        "ToPort": 443
                    }
                ],
                "Tags": [
                    {
                        "Key": "KubernetesCluster",
                        "Value": {
                            "Ref": "ClusterName"
                        }
                    }
                ],
                "VpcId": {
                    "Ref": "VPC"
                }
            },
            "Type": "AWS::EC2::SecurityGroup"
        },
        "SecurityGroupControllerIngressFromWorkerToEtcd": {
            "Properties": {
                "FromPort": 2379,
                "GroupId": {
                    "Ref": "SecurityGroupController"
                },
                "IpProtocol": "tcp",
                "SourceSecurityGroupId": {
                    "Ref": "SecurityGroupWorker"
                },
                "ToPort": 2379
            },
            "Type": "AWS::EC2::SecurityGroupIngress"
        },
        "SecurityGroupWorker": {
            "Properties": {
                "GroupDescription": {
                    "Ref": "AWS::StackName"
                },
                "SecurityGroupEgress": [
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 0,
                        "IpProtocol": "tcp",
                        "ToPort": 65535
                    },
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 0,
                        "IpProtocol": "udp",
                        "ToPort": 65535
                    }
                ],
                "SecurityGroupIngress": [
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 3,
                        "IpProtocol": "icmp",
                        "ToPort": -1
                    },
                    {
                        "CidrIp": "0.0.0.0/0",
                        "FromPort": 22,
                        "IpProtocol": "tcp",
                        "ToPort": 22
                    }
                ],
                "Tags": [
                    {
                        "Key": "KubernetesCluster",
                        "Value": {
                            "Ref": "ClusterName"
                        }
                    }
                ],
                "VpcId": {
                    "Ref": "VPC"
                }
            },
            "Type": "AWS::EC2::SecurityGroup"
        },
        "SecurityGroupWorkerIngressFromControllerToFlannel": {
            "Properties": {
                "FromPort": 8285,
                "GroupId": {
                    "Ref": "SecurityGroupWorker"
                },
                "IpProtocol": "udp",
                "SourceSecurityGroupId": {
                    "Ref": "SecurityGroupController"
                },
                "ToPort": 8285
            },
            "Type": "AWS::EC2::SecurityGroupIngress"
        },
        "SecurityGroupWorkerIngressFromControllerToKubelet": {
            "Properties": {
                "FromPort": 10250,
                "GroupId": {
                    "Ref": "SecurityGroupWorker"
                },
                "IpProtocol": "tcp",
                "SourceSecurityGroupId": {
                    "Ref": "SecurityGroupController"
                },
                "ToPort": 10250
            },
            "Type": "AWS::EC2::SecurityGroupIngress"
        },
        "SecurityGroupWorkerIngressFromControllerTocAdvisor": {
            "Properties": {
                "FromPort": 4194,
                "GroupId": {
                    "Ref": "SecurityGroupWorker"
                },
                "IpProtocol": "tcp",
                "SourceSecurityGroupId": {
                    "Ref": "SecurityGroupController"
                },
                "ToPort": 4194
            },
            "Type": "AWS::EC2::SecurityGroupIngress"
        },
        "SecurityGroupWorkerIngressFromWorkerToFlannel": {
            "Properties": {
                "FromPort": 8285,
                "GroupId": {
                    "Ref": "SecurityGroupWorker"
                },
                "IpProtocol": "udp",
                "SourceSecurityGroupId": {
                    "Ref": "SecurityGroupWorker"
                },
                "ToPort": 8285
            },
            "Type": "AWS::EC2::SecurityGroupIngress"
        },
        "SecurityGroupWorkerIngressFromWorkerToKubeletReadOnly": {
            "Properties": {
                "FromPort": 10255,
                "GroupId": {
                    "Ref": "SecurityGroupWorker"
                },
                "IpProtocol": "tcp",
                "SourceSecurityGroupId": {
                    "Ref": "SecurityGroupWorker"
                },
                "ToPort": 10255
            },
            "Type": "AWS::EC2::SecurityGroupIngress"
        },
        "Subnet": {
            "Properties": {
                "AvailabilityZone": {
                    "Fn::If": [
                        "EmptyAvailabilityZone",
                        {
                            "Fn::Select": [
                                "0",
                                {
                                    "Fn::GetAZs": {
                                        "Ref": "AWS::Region"
                                    }
                                }
                            ]
                        },
                        {
                            "Ref": "AvailabilityZone"
                        }
                    ]
                },
                "CidrBlock": "10.0.0.0/24",
                "MapPublicIpOnLaunch": true,
                "Tags": [
                    {
                        "Key": "KubernetesCluster",
                        "Value": {
                            "Ref": "ClusterName"
                        }
                    }
                ],
                "VpcId": {
                    "Ref": "VPC"
                }
            },
            "Type": "AWS::EC2::Subnet"
        },
        "SubnetRouteTableAssociation": {
            "Properties": {
                "RouteTableId": {
                    "Ref": "RouteTable"
                },
                "SubnetId": {
                    "Ref": "Subnet"
                }
            },
            "Type": "AWS::EC2::SubnetRouteTableAssociation"
        },
        "VPC": {
            "Properties": {
                "CidrBlock": "10.0.0.0/16",
                "EnableDnsHostnames": true,
                "EnableDnsSupport": true,
                "InstanceTenancy": "default",
                "Tags": [
                    {
                        "Key": "KubernetesCluster",
                        "Value": {
                            "Ref": "ClusterName"
                        }
                    },
                    {
                        "Key": "Name",
                        "Value": "kubernetes-vpc"
                    }
                ]
            },
            "Type": "AWS::EC2::VPC"
        },
        "VPCGatewayAttachment": {
            "Properties": {
                "InternetGatewayId": {
                    "Ref": "InternetGateway"
                },
                "VpcId": {
                    "Ref": "VPC"
                }
            },
            "Type": "AWS::EC2::VPCGatewayAttachment"
        }
    }
}
```

**Comment by jeffb-stell** [comment]
This may be an issue of new functionality being prematurely exposed? https://github.com/coreos/coreos-kubernetes/commit/11e75bf0162790f557f4a76a34e7ac7b44a40f76#diff-da222392c29da368a3a2c580c10576e2 indicates that the enhancements to the rendered cfn template to allow setting subnetting parameters went in after the 0.3.0 release.

Any chance of cutting a new release?

**Comment by rclayton-the-terrible** [comment]
I should add that this occurs both in the current release and on master (built from source).

**Comment by xied75** [comment]
Dear all, unless I got everything wrong, this is not a bug at all. The aws cloudformation template does not change as you hoped that it will merge the values from your yaml. No the template is submitted as is, and whatever you specified is supplied to aws cloudformation as parameter values, that's how they work. If you go aws console and look at cloudformation stack you'll see your parameters.

Given the trend to use immutable resources, the template is FIXED, what you supply is run-time information, thus the parameter stuff.

**Comment by xied75** [comment]
Now a related issue is what does "kube-aws render" really do, it looks like it just grabbed the file from s3 storage, without doing anything extra, you can do the same with curl or awscli in one line too.

**Comment by jeffb-stell** [comment]
@xied75 - the AWS CFN template in use (the one hosted in S3 by CoreOS, and the one generated by kube-aws render) does not use any passed subnetting parameters when defining network (VPC, subnet, etc) resources. That is the primary cause of the problem, seemingly.

Also, that's not what kube-aws render does. It creates (using a golang cfn library) a CloudFormation template suitable for upload to S3, to be referenced by the kube-aws up command.

**Comment by rclayton-the-terrible** [comment]
@xied75 Yeah, I have to confirm what @jeffb-stell said.  I spent a good 4 hours compiling a custom version of kube-aws to ensure the correct values where being supplied (and they weren't).  I could only get the thing to work by changing the default values in the Golang source to those I needed for deployment.

**Comment by colhom** [comment]
We have #240 in review now. Should land soon.

TLDR; `kube-render` will generate all artifacts (now called assets), and `kube-up` will use those on-disk assets to create the cloudformation. No more S3 intermediary for the artifacts- on `kube-aws up` everything is pulled from the asset directory created by `kube-aws render`

**Comment by allingeek** [comment]
This all leaves me to wonder, "What is the point of CIDR prompts in the yaml if the network provisioned in AWS is always the same 10.0.0.0/24?"

**Comment by colhom** [comment]
As mentioned above, you must use the template.json file created by `kube-aws render` in order to set the VPC CIDR (as well as the rest of the networking configuration options that have become available since the last release)

We are in the process of cutting a new release, which will change (read: simplify) the way that assets are templated into cloudformation stack. Until then, if you want to use the network configuration stuff you'll have to render your own template.json file and host it in an s3 bucket.

**Comment by allingeek** [comment]
Yes. Sorry for posting the incomplete thought. The point is that if so many people are making this same mistake then maybe it'd be best to add something like "INTERNAL" or "VIRTUAL" or "OVERLAY" in the comments above networking related fields in the yaml.

**Comment by jeffb-stell** [comment]
Note that the template generated by the render command is also incorrect
 (does not incorporate CIDR values)

**Comment by colhom** [comment]
@jeffb-stell This is not a bug. The cloudformation template only includes the default values for parameters. _Even when you render your own template with custom networking parameter values, template.json will only contain the default values_. The `kube-aws up` command provides the custom parameters on the API call for stack creation. 

The reason users are experiencing this problem while using the default coreos-hosted `artifactUrl` is that the v0.3.0 cloudformation template does not include _references to the new networking parameters_. This means that when `kube-aws up` is run, the networking parameters simply do not exist in the template and are ignored. This is certainly un-intuitive, and it's easy to see how folks could get confused.

As of tonight, the v0.4.0 release has been cut. The kube-aws binary packaged with that release will by default use the newly uploaded v0.4.0 artifact url, which means that networking configuration works "out of the box" now, and does not require rendering/hosting your own template.

**Comment by rclayton-the-terrible** [comment]
@colhom - I would say that it is not obvious what the intent of the `render` command is if it does not actually show what the environment would look like with my environment-specific settings applied.  I understand that this is the nature of CloudFormation, but I have to say it seems quite unintuitive.

**Comment by colhom** [comment]
@rclayton-the-terrible I agree completely. We're currently working through a major revamp of this tool which will alleviate this nasty tangled web of _things parameterizing things which parameterize other things...._.
